sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -0,0 +1,181 @@
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import AzureCommunication
+import AzureCore
+import Foundation
+
+public class ChatClient {
+    // MARK: Properties
+
+    private let client: AzureCommunicationChatClient
+    private let credential: CommunicationUserCredential
+    private let endpoint: String
+    private let options: AzureCommunicationChatClientOptions
+
+    // MARK: Initializers
+
+    /// Create a ChatClient.
+    /// - Parameters:
+    ///   - endpoint: The Communication Services endpoint.
+    ///   - credential: The user credential.
+    ///   - options: Options used to configure the client.
+    public init(
+        endpoint: String,
+        credential: CommunicationUserCredential,
+        withOptions options: AzureCommunicationChatClientOptions
+    ) throws {
+        self.endpoint = endpoint
+        self.credential = credential
+        self.options = options
+
+        guard let baseUrl = URL(string: endpoint) else {
+            throw AzureError.client("Unable to form base URL.")
+        }
+
+        let authPolicy = CommunicationUserCredentialPolicy(credential: credential)
+
+        self.client = try AzureCommunicationChatClient(baseUrl: baseUrl, authPolicy: authPolicy, withOptions: options)
+    }
+
+    // MARK: Public Methods
+
+    /// Get a ChatThreadClient.
+    /// - Parameters:
+    ///   - threadId: The threadId.
+    public func getChatThreadClient(threadId: String) throws -> ChatThreadClient {
+        return try ChatThreadClient(
+            threadId: threadId,
+            endpoint: endpoint,
+            credential: credential,
+            withOptions: options
+        )
+    }
+
+    /// Create a ChatThread.
+    /// - Parameters:
+    ///   - request: Request for creating a chat thread with the topic and members to add.
+    ///   - withOptions: Create chat thread options.
+    ///   - completionHandler: A completion handler that receives a ChatThreadClient on success.
+    public func createChatThread(
+        request: CreateChatThreadRequest,
+        withOptions options: CreateChatThreadOptions? = nil,
+        completionHandler: @escaping (Result<ChatThreadClient, Error>) -> Void
```

Comment:
Why is this create call returning a `ChatThreadClient` instead of a `ChatThread`? This seems very unusual. The "normal" case would be to create a `ChatThread` and then create (or not create!) a `ChatThreadClient`.

This should not return `Result<T>` but instead `HTTPResultHandler<T>` so that the httpResponse can be passed through to the calling app, if desired. 

https://azure.github.io/azure-sdk/ios_design.html#ios-network-callback-type